packages/grid/x-data-grid/src/hooks/features/filter/gridFilterUtils.ts

```diff
@@ -354,18 +372,28 @@ export const buildAggregatedQuickFilterApplier = (
     if (getApplyQuickFilterFnV7 && !(hasUserFunctionLegacy && !hasUserFunctionV7)) {
       appliersPerField.push({
         column,
-        appliers: quickFilterValues.map((value) => ({
-          v7: true,
-          fn: getApplyQuickFilterFnV7(value, column, apiRef),
-        })),
+        appliers: quickFilterValues.map((quickFilterValue) => {
+          const value = column.ignoreDiacritics
```

Comment:
Doesn't necessarily just apply to string columns. Not sure what's the default date format but e.g. "February" is "février" in French so we can get diacritics elsewhere, and those are supposed to be searchable through the quickfilter. Any column can return a formatted localized (aka with diacritics) value with the `.valueFormatter` also.

I think it might make sense to make it a root-level option.

Reply:
Indeed, for quick filters, it could apply to any column type.
For regular filters and header filters, it will only apply to the string column type.

I was thinking about where to put this option and decided to put it in `GridColDef` to allow per-column configuration.
But it's a bit of a pain if you have a lot of columns and want to ignore diacritics in all of them.
This seems to emphasize a larger problem - we don't provide a way to pass something to all columns at once (or all columns of a certain type at once).

Regarding `ignoreDiacritics` specifically, enabling it on the root level should cover most, if not all, use cases.
So this makes sense to me, I will update the demo and the docs 👍🏻